src/java.base/share/classes/java/nio/file/Path.java

```diff
@@ -248,6 +256,55 @@ public static Path of(URI uri) {
      */
     Path getFileName();
 
+    /**
+     * Returns the file name extension of this path as a {@code String}. The
+     * extension is defined to be the portion of the {@code String}
+     * representation of the file name after the last dot ('.'). All leading
+     * dots are ignored. If the extension is missing, then an
+     * {@link String#isEmpty() empty} {@code String} ({@code ""}) is returned.
```

Comment:
Don’t you mean to say here that if the extension is missing then `null` is returned? That’s what the `@return` tag below implies, and that’s what would be most useful.

Reply:
Yes, that is what it should be. This was translated incorrectly from the previous version returning `Optional<String>`.

Reply:
Fixed in 8a4a4bc6fc41453d9480a5eba0a8f227dd7ceadc.